### PR TITLE
Fix image processing

### PR DESCRIPTION
**Pull request by Blaussh**

### CODE REVIEW

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 1]
```diff
@@ -138,6 +138,31 @@ export class ToolsAppEffects {
...
+	 */
+	@Effect()
+	onActiveMapChangesUpdateHash$: Observable<any> = this.actions$
+		.ofType(MapActionTypes.TRIGGER.ACTIVE_MAP_CHANGED)
+		.withLatestFrom(this.store$.select(toolsStateSelector), this.store$.select(mapStateSelector))
+		.filter(([action, toolsState, mapState]: [ActiveMapChangedAction, IToolsState, IMapState]) => toolsState.imageProcessingHash.hasOwnProperty(action.payload))
+		.mergeMap(([action, toolsState, mapState]: [ActiveMapChangedAction, IToolsState, IMapState]) => {
+			const activeMap: CaseMapState = MapFacadeService.activeMap(mapState);
+			if (!activeMap.data.overlay) {
+				return [new DisableImageProcessing()];
```
Comment (shaharido1) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: CONSOLE.LOG IF IT'S POSSIBLE

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 2]
```diff
@@ -138,6 +138,31 @@ export class ToolsAppEffects {
...
+	/**
+	 * @type Effect
+	 * @name onActiveMapChangesUpdateHash$
+	 * @ofType ActiveMapChangedAction
+	 * @dependencies map
+	 * @filter
+	 * @action DisableImageProcessing?, EnableImageProcessing?, SetManualImageProcessingArguments?
+	 */
+	@Effect()
+	onActiveMapChangesUpdateHash$: Observable<any> = this.actions$
```
Comment (shaharido1) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: CHANGE NAME

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 3]
```diff
@@ -183,11 +208,23 @@ export class ToolsAppEffects {
 		.mergeMap(([action, selectedMap, selectedCase]: [DisplayOverlaySuccessAction, CaseMapState, Case]) => {
 			// action 1: EnableImageProcessing
 			const actions = [new EnableImageProcessing()];
-			let manualProcessArgs;
+			let manualProcessArgs: ImageManualProcessArgs;
+			let params: Array<IImageProcParamComp> = this.config.ImageProcParams.map(param => {
```
Comment (shaharido1) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: fix map function

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 4]
```diff
@@ -183,11 +208,23 @@ export class ToolsAppEffects {
...
+				Brightness: params[2].defaultValue,
+				Gamma: params[3].defaultValue,
+				Saturation: params[4].defaultValue
+			};
 
 			// action 2: SetMapManualImageProcessing / SetMapAutoImageProcessing (optional)
 			if (selectedCase.state.overlaysManualProcessArgs) {
-				manualProcessArgs = selectedCase.state.overlaysManualProcessArgs[action.payload.overlay.id];
+				manualProcessArgs = Boolean(selectedCase.state.overlaysManualProcessArgs[action.payload.overlay.id]) ?
+					selectedCase.state.overlaysManualProcessArgs[action.payload.overlay.id] : manualProcessArgs;
```
Comment (shaharido1) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: selectedCase.state.overlaysManualProcessArgs[action.payload.overlay.id] || manualProcessArgs

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts (hunk 1 above)
Comment (Blaussh) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: It is possible, the check is needed

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts (hunk 4 above)
Comment (Blaussh) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: Done

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts (hunk 3 above)
Comment (Blaussh) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: Done

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts (hunk 2 above)
Comment (Blaussh) [reply src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: Done

**Pull request merged**